src/lib.rs

```diff
@@ -1233,6 +1233,14 @@ impl Device {
     /// Creates a new buffer, maps it into host-visible memory, copies data from the given slice,
     /// and finally unmaps it, returning a [`Buffer`].
     pub fn create_buffer_with_data(&self, data: &[u8], usage: BufferUsage) -> Buffer {
+        let mut data = Cow::Borrowed(data);
```

Comment:
can we write this without cloning the whole data vector? it may be costy!

Reply:
How about this?
I could do a copy_from_slice from a constant [0; 4] but I dont think its worth the trouble?

I tested the changes on my game (which breaks when alignment isnt handled) and it works fine.